iree/compiler/Conversion/Common/Transforms.h

```diff
@@ -103,6 +104,24 @@ LogicalResult tileAndFuseLinalgBufferOps(
     const linalg::LinalgDependenceGraph &dependenceGraph,
     const LaunchConfig &launchConfig, const TileAndFuseOptions &options);
 
+/// Perform folding of AffineMinOp by matching the pattern generated by tile and
+/// distribute.
+/// Try to fold a affine.min op by matching the following form:
+/// ```
+/// scf.for %iv = %lb to %ub step %step
+///   %affine.min affine_map<(d0, d1) -> (N, d0 - d1)>(%ub, %iv)
+/// ```
+/// With N a compile time constant. This operations can be replace by
+/// `%cN = constant N : index` if we can prove that %lb, %step and %ub are
+/// divisible by N.
+struct AffineMinDistributedSCFCanonicalizationPattern
```

Comment:
I'd rather not expose the struct directly, but rather use an "populateAffineMinSCFCanonicalizationPattern" method.

Reply:
Make sense, I moved the structure into the cpp file